src/docfx/build/metadata/MetadataProvider.cs

```diff
@@ -48,7 +48,13 @@ public JObject GetMetadata(Document file, JObject yamlHeader = null)
             {
                 result.Merge(yamlHeader, JsonUtility.MergeSettings);
             }
-            return result;
+
+            if (result.TryGetValue("redirect_url", StringComparison.OrdinalIgnoreCase, out _))
+            {
+                return (Errors.RedirectionInMetadata(), result);
```

Comment:
This could be solved in a general way with a more general error code. There are more metadatas to block in addition to `redirect_url`, like `monikers`. I'd like to see a way to _attribute_ these blacklisted metadata.